norman_core/clients/http_client.py

```diff
@@ -74,7 +74,15 @@ def _create_headers(self, token: Optional[Sensitive[str]]):
 
     @staticmethod
     def _parse_response(response: httpx.Response, response_encoding: ResponseEncoding):
-        response.raise_for_status()
+        try:
+            response.raise_for_status()
+        except httpx.HTTPStatusError as e:
+            try:
+                detail = e.response.json()
+            except Exception:
+                detail = e.response.text
+            message = f"HTTP {e.response.status_code} {e.request.url}: {detail}"
```

Comment:
Design: this is public code. Let's structure the error into something that respects our company and public image.